src/ol_infrastructure/applications/mitopen/Pulumi.applications.mitopen.QA.yaml

```diff
@@ -3,6 +3,45 @@ secretsprovider: awskms://alias/infrastructure-secrets-qa
 encryptedkey: AQICAHg42pDDDGBhpaX14TdtzcK1hbiMYTHsYRH4k5GL5RFpIwEi8MniqKSK7PJ7AIbp17REAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMbWm9PEFY6H+jKGYHAgEQgDtAZs04prjkVBGB8O8+C6kjqzMhazPYOAanmLoWdx4WArJXNKxZkv72c6my81/qnCg0xWYdmw6fJxV0dg==
 config:
   consul:address: https://consul-apps-qa.odl.mit.edu
+  heroku:user: "odl-devops"
+  heroku:app_id:
+    secure: v1:eSvnqne7KQRkHJlu:wTyl9W3aivfxlANqBML2wy4GEHe85VuKll60R5HmGUu1hJTcQBEfzK1cJOyvjlbLgBtXLw==
+  heroku_var:celery_worker_max_memory_per_child: 125000
+  heroku_var:cors_urls:
+  - "https://ocwnext-rc.odl.mit.edu"
+  - "https://ocw-next.netlify.app"
+  - "https://ol-devops-ci.odl.mit.edu"
+  - "https://draft-qa.ocw.mit.edu"
+  - "https://live-qa.ocw.mit.edu"
+  heroku_var:debug: false
+  heroku_var:edx_learning_course_bucket_name: "edxorg-qa-edxapp-courses"
+  heroku_var:enable_infitite_corridor: true
+  heroku_var:etl_micromasters_host: "micromasters-rc.odl.mit.edu"
+  heroku_var:etl_xpro_host: "rc.xpro.mit.edu"
+  heroku_var:ga_g_tracking_id: ""
+  heroku_var:ga_tracking_id: ""
+  heroku_var:indexing_api_username: "od_mm_rc_api"
+  heroku_var:keycloak_base_url: "https://sso-qa.ol.mit.edu"
+  heroku_var:keycloak_realm_name: "olapps"
+  heroku_var:log_level: "INFO"
+  heroku_var:mailgun_sender_domain: "discussions-mail.odl.mit.edu"
+  heroku_var:mitopen_base_url: "https://mit-open-rc.odl.mit.edu"
+  heroku_var:mitopen_cookie_domain: "odl.mit.edu"
+  heroku_var:mitopen_cookie_name: "mitopen-rc"
+  heroku_var:mitopen_support_email: "odl-mitopen-rc-support@mit.edu"
+  heroku_var:ocw_base_url: "https://live-qa.ocw.mit.edu/"
+  heroku_var:ocw_content_bucket_name: "ocw-content-storage"
+  heroku_var:ocw_iterator_chunk_size: 300
+  heroku_var:ocw_live_bucket: "ocw-content-live-qa"
+  heroku_var:ocw_next_aws_storage_bucket_name: "ol-ocw-studio-app-qa"
+  heroku_var:opensearch_index: "mitxopen-rc"
+  heroku_var:opensearch_shard_count: 2
+  heroku_var:opensearch_url: "https://search-opensearch-mitopen-qa-nvwau2pbr2nek5dgx2o5w3xju4.us-east-1.es.amazonaws.com"
+  heroku_var:pgbouncer_default_pool_size: 50
+  heroku_var:pgbouncer_max_client_conn: 500
+  heroku_var:pgbouncer_min_pool_size: 20
+  heroku_var:sso_url: "sso-qa.ol.mit.edu"
+  heroku_var:tika_server_endpoint: "https://tika-qa.odl.mit.edu"
```

Comment:
This might be more straightforward as something like 
```
heroku_app:vars:
  CORS_ALLOWED_ORIGINS:
  - "https://..."
  ENABLE_INFINITE_CORRIDOR: true
```

And then in the Pulumi code use it like:

```
heroku_app_vars = Config("heroku_app").get_object("vars")
heroku_vars = {...}
heroku_vars.update(**heroku_app_vars)
```

Reply:
I think something like that would work well if it were just a 1-1 mapping of this config -> rendered config but I think it would result in something even more convoluted for lines like these where the value is interpolated into strings. Unless I'm missing something obvious, which I probably am. 

```
    "MAILGUN_FROM_EMAIL": f"MIT Open <no-reply@{heroku_var_config.require('mailgun_sender_domain')}",
    "MAILGUN_SENDER_DOMAIN": heroku_var_config.require("mailgun_sender_domain"),
    "MAILGUN_URL": f"https://api.mailgun.net/v3/{heroku_var_config.require('mailgun_sender_domain')}",
```

Reply:
That's a fair point. We can do a mix though, where a majority of values can be specified as an object, and anything that requires interpolation/shared data can be custom coded. I wouldn't want to jam the entire config object into the stack config, but the dict based approach reduces the amount of boilerplate that is needed to add/remove new values.